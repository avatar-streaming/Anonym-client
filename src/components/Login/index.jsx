import React from "react";
import useLogin from "../../hooks/useLogin";

function Login() {
  const handleLogin = useLogin();

  return (
    <div className="login-container">
      <div className="app-title">
        Anonym
      </div>
      <button className="login-button" onClick={() => handleLogin(true)}>
        Goggle Login
      </button>
    </div>
  );
}

export default Login;
