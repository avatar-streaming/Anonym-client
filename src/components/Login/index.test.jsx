import React from "react";
import Login from ".";
import renderer from "react-test-renderer";

describe("<Login />", () => {
  let component = null;

  beforeEach(() => {
    component = renderer.create(<Login />);
  });

  it("should render with given state from redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should login", () => {
    const loginButton = component.root.findByProps({ "data-testid": "login" });
    let count = 0;

    expect(count).toEqual(0);

    renderer.act(() => {
      loginButton.props.onClick({
        buttonCount: count++
      });
    });
    expect(count).toEqual(1);
  });
});
