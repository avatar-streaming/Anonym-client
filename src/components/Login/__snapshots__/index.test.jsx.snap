// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render with given state from redux store 1`] = `
<div
  className="login-container"
>
  <div
    className="login-container__title"
  >
    Anonym
  </div>
  <button
    className="login-container__login"
    data-testid="login"
    onClick={[Function]}
  >
    Goggle Login
  </button>
</div>
`;
